settings/request.go

```diff
@@ -0,0 +1,278 @@
+//	+----------------------------+-------------------------+---------+---------------------------+
+//	|       SQLParserDebug       | Defines whether to show |  false  | Since 2.3.1 (only if      |
+//	|                            | parser steps for        |         | built with                |
+//	|                            | following statements.   |         | -DCMAKE_BUILD_TYPE=Debug) |
+//	+----------------------------+-------------------------+---------+---------------------------+
+//	|    SQLRecursiveTriggers    | Defines whether a       |  true   | Since 2.3.1.              |
+//	|                            | triggered statement can |         |                           |
+//	|                            | activate a trigger.     |         |                           |
+//	+----------------------------+-------------------------+---------+---------------------------+
+//	| SQLReverseUnorderedSelects | Defines defines whether |  false  | Since 2.3.1.              |
+//	|                            | result rows are usually |         |                           |
+//	|                            | in reverse order if     |         |                           |
+//	|                            | there is no ORDER BY    |         |                           |
+//	|                            | clause.                 |         |                           |
+//	+----------------------------+-------------------------+---------+---------------------------+
+//	|       SQLSelectDebug       | Defines whether to show |  false  | Since 2.3.1 (only if      |
+//	|                            | to show execution steps |         | built with                |
+//	|                            | during SELECT.          |         | -DCMAKE_BUILD_TYPE=Debug) |
+//	+----------------------------+-------------------------+---------+---------------------------+
+//	|        SQLVDBEDebug        | Defines whether VDBE    |  false  | Since 2.3.1 (only if      |
+//	|                            | debug mode is enabled.  |         | built with                |
+//	|                            |                         |         | -DCMAKE_BUILD_TYPE=Debug) |
+//	+----------------------------+-------------------------+---------+---------------------------+
+//
+// Since: 1.10.0
+//
+// See also:
+//
+// * Session settings https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/_session_settings/
+package settings
+
+import (
+	"context"
+
+	"github.com/tarantool/go-tarantool"
+)
+
+// Request helps to work with session settings.
+type Request struct {
+	get *tarantool.SelectRequest
+	set *tarantool.UpdateRequest
+}
+
+func (req *Request) assertBuiltWithNew() {
+	if (req.get == nil) && (req.set == nil) {
+		panic("Invalid settings.Request (empty), possibly build not with New*Request function")
+	}
+
+	if (req.get != nil) && (req.set != nil) {
+		panic("Invalid settings.Request (both types), possibly build not with New*Request function")
+	}
+}
```

Comment:
I would not add such checks. In any case, it will fall with a panic (SIGSEGV).

Reply:
Removed in rework